src/components/icon-button/index.js

```diff
@@ -23,6 +24,7 @@ const IconButton = ({
   return visible ? (
     <Container>
       <button
+        data-heap-event={dataHeapEvent}
```

Comment:
Given that this relies on the prop that is passed down from `iconButton`, we would need a test to indicate the right type of prop gets passed in  from the metadata panel for this heap event.